main.go

```
@@ -3,208 +3,257 @@ package main
+				set = setCmd - cas
 			} else {
-				e.bytes.WithLabelValues(dir).Set(float64(m))
+				log.Errorf("Failed to parse cas: %s", casErr)
 			}
+		} else {
+			log.Errorf("Failed to parse set %q: %s", s["cmd_set"], err)
 		}
+		ch <- prometheus.MustNewConstMetric(e.commands, prometheus.CounterValue, set, "set", "hit")
 
-		for _, st := range removalsStatuses {
-			m, err := strconv.ParseUint(stats[server][fmt.Sprintf("%s_unfetched", st)], 10, 64)
-			if err != nil {
-				e.removals.WithLabelValues(st, "unfetched").Set(0)
-			} else {
-				e.removals.WithLabelValues(st, "unfetched").Set(float64(m))
-			}
-		}
-		m, err = strconv.ParseUint(stats[server]["evicted"], 10, 64)
+		ch <- prometheus.MustNewConstMetric(e.currentBytes, prometheus.GaugeValue, parse(s, "bytes"))
+		ch <- prometheus.MustNewConstMetric(e.limitBytes, prometheus.GaugeValue, parse(s, "limit_maxbytes"))
+		ch <- prometheus.MustNewConstMetric(e.items, prometheus.GaugeValue, parse(s, "curr_items"))
+		ch <- prometheus.MustNewConstMetric(e.itemsTotal, prometheus.CounterValue, parse(s, "total_items"))
+
+		ch <- prometheus.MustNewConstMetric(e.bytesRead, prometheus.CounterValue, parse(s, "bytes_read"))
+		ch <- prometheus.MustNewConstMetric(e.bytesWritten, prometheus.CounterValue, parse(s, "bytes_written"))
+
+		ch <- prometheus.MustNewConstMetric(e.connections, prometheus.GaugeValue, parse(s, "curr_connections"))
+		ch <- prometheus.MustNewConstMetric(e.connectionsTotal, prometheus.CounterValue, parse(s, "total_connections"))
+
+		ch <- prometheus.MustNewConstMetric(e.evictions, prometheus.CounterValue, parse(s, "evictions"))
+		ch <- prometheus.MustNewConstMetric(e.reclaimed, prometheus.CounterValue, parse(s, "reclaimed"))
+	}
+
+}
+
+func parse(stats map[string]string, key string) float64 {
+	v, err := strconv.ParseFloat(stats[key], 64)
+	if err != nil {
+		log.Errorf("Failed to parse %s %q: %s", key, stats[key], err)
+		v = 0
```

This could be misleading and hide errors - would it be better to `NaN` instead?